### PR TITLE
openthread: doc: describe there are multiple libraries for OpenThread

### PR DESCRIPTION
**Pull request by MarekPorwisz**
Added info there are variants depending on stack and crypto.
Described Configuration used to prepeare each variant
Added info how to select stack variants

Signed-off-by: Marek Porwisz <marek.porwisz@nordicsemi.no>

- [x] blocked by nrfconnect/sdk-nrf#2974

### CODE REVIEW

**Comment by MarekPorwisz** [comment]
Please have a look and let me know If there is something more needed. 
Note that I also removed settings related to ID/network name/channel/etc as those are fully configurable and shall not impact certification

File: openthread/README.rst
Comment (edmont) [inline openthread/README.rst:21]: Commit hash should be updated, right?

File: openthread/README.rst
Comment (edmont) [inline openthread/README.rst:35]: What about center-aligning these columns?

File: openthread/README.rst
Comment (edmont) [inline openthread/README.rst:193]: Use commas to separate options or better a list.

**Review by edmont — COMMENTED** [review]
Wait for https://github.com/nrfconnect/sdk-nrfxlib/pull/268 before merging?

File: openthread/README.rst
Comment (MarekPorwisz) [reply openthread/README.rst:35]: rst does not support alignment in tables.

File: openthread/README.rst
Comment (MarekPorwisz) [reply openthread/README.rst:21]: I don't have the commit yet:) As you mentioned this will be updated after #268

**Comment by greg-fer** [comment]
Reviewed -- please check my commit. As mentioned to Marek and Lukasz, I don't believe we should mention how to use this libraries explicitly, but link to the related section in Configuring Thread in NCS.

Also, for me this PR is blocked by https://github.com/nrfconnect/sdk-nrf/pull/2974

**Review by greg-fer — APPROVED** [review]

File: openthread/README.rst
Comment (lmaciejonczyk) [inline openthread/README.rst:20]:
```suggestion
The OpenThread pre-built libraries were created based on commit ``07f430dac6c400abdd358ba1173c0f94b94a5a80`` of OpenThread,
```

**Review by lmaciejonczyk — CHANGES_REQUESTED** [review]
The OpenThread commit differs

File: openthread/README.rst
Comment (MarekPorwisz) [reply openthread/README.rst]: updated

**Review by lmaciejonczyk — APPROVED** [review]

**Review by edmont — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Pull request merged**
